src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1660,7 +1668,26 @@ private static HttpClient CreateHttpClient(HttpClientParameters parameters, Http
 
                     try
                     {
-                        socket.NoDelay = true;
+                        if (parameters.ServicePoint is not null)
+                        {
+                            socket.NoDelay = !parameters.ServicePoint.UseNagleAlgorithm;
```

Comment:
ServicePoint originally used the `Nagle Algorithm`. The default value for the `UseNagleAlgorithm` is `true`.
So, if we set this property to `false` underlying Socket will use the `Nagle Algorithm` as well. It's mostly because of consistency with the .NET Framework

Reply:
I feel the `socket.NoDelay = true` should be default. Nagle is harmful to HTTP IMHO.

Reply:
I left it as-is in that case.